app/src/main/java/com/github/muellerma/prepaidbalance/work/CheckBalanceWorker.kt

```diff
@@ -155,8 +156,13 @@ class CheckBalanceWorker(
                 return callback(CheckResult.USSD_INVALID, null)
             }
 
+            val subscriptionId = context.prefs().getString("subscription_id", "")
+                .orEmpty().toIntOrNull() ?: return callback(CheckResult.SUBSCRIPTION_INVALID, null)
```

Comment:
Can you handle `subscriptionId  == null` as "use default sim card", so don't call `createForSubscriptionId()` below. Then you can remove `setDefaultSubscriptionId()` and maybe also remove `CheckResult.SUBSCRIPTION_INVALID`.

Reply:
@mueller-ma, if you don't use `createForSubscriptionId()`, then while using 2 sim cards, you will get same error as before.
This PR was intended to fix that error. So I think we can't ignore  `createForSubscriptionId()`.
What else do you propose while keeping this limitation?

Reply:
I want to ensure that there's no error for people with one sim card. If a user has multiple sim cards, he can set the correct one in the settings.